builtin/logical/database/backend.go

```diff
@@ -27,6 +29,29 @@ const (
 	minRootCredRollbackAge = 1 * time.Minute
 )
 
+// metrics collection
+var (
+	gaugeSync = sync.Mutex{}
+	gauges    = map[string]*int32{}
+	gaugeKey  = []string{"secrets", "database", "backend", "connections", "count"}
+)
+
+func createGauge(name string) {
+	gaugeSync.Lock()
+	defer gaugeSync.Unlock()
+	gauges[name] = new(int32)
+}
+
+func addToGauge(dbType, version string, uuid string, amount int32) {
```

Comment:
For consistency, either:

```suggestion
func addToGauge(dbType, version, uuid string, amount int32) {
```

Or:

```suggestion
func addToGauge(dbType string, version string, uuid string, amount int32) {
```